scapy/layers/dns.py

```diff
@@ -127,11 +151,10 @@ def possible_shortens(dat):
             yield dat.split(b".", x)[x]
     data = {}
     burned_data = 0
-    dummy_dns = DNSStrField("", "")  # Used for its i2m method
     for current, name, dat in field_gen(dns_pkt):
         for part in possible_shortens(dat):
             # Encode the data
-            encoded = dummy_dns.i2m(None, part)
+            encoded = dns_encode(part, check_built=True)
```

Comment:
I hadn't noticed before :-( The part may also be an rdata string, which doesn't have the any2i that adds the final dot. So...
```
>>> raw(DNS(an=DNSRR(rrname='a\u0080', type=2, rdata='b\u0080')))
b'\x00\x00\x01\x00\x00\x00\x00\x01\x00\x00\x00\x00\x03a\xc2\x80\x00\x00\x02\x00\x01\x00\x00\x00\x00\x00\x03b\xc2\x80'
                                                  ^^^^^^^^^^^^^^^^^encoded                                 ^^^^^^^^^not encoded
```

So, RDataField should also get an any2i that adds the final dot if it is missing.

Which makes me think: maybe the encoding can be done in any2i directly?

Reply:
Doing the encoding in `any2i` would display the encoded bytes instead of a human-readable value when performing `.show()` for instance.